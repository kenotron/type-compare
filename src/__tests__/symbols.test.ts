import { test } from "@jest/globals";
import { comapreHarness, getCasesByGroup } from "./compare-harness";

const group = "symbols";

test.concurrent.each(getCasesByGroup(group))(`Test case "%s"`, async (testCase) => {
  comapreHarness(group, testCase);
});
