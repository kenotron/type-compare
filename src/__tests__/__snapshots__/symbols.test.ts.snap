// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test case "added-symbol" 1`] = `
{
  "messages": [],
  "minChangeType": "none",
}
`;

exports[`Test case "namespace-to-object-type" 1`] = `
{
  "messages": [
    "[symbol-kind-changed] Export "F" symbol has changed (namespace, class, interface, or function).",
  ],
  "minChangeType": "major",
}
`;

exports[`Test case "removed-symbol" 1`] = `
{
  "messages": [
    "[no-symbol-name-removed] Export "Bar" was removed from export class Foo {}
",
  ],
  "minChangeType": "major",
}
`;

exports[`Test case "symbol-add-export-type" 1`] = `
{
  "messages": [
    "[symbol-kind-changed] Export "F" symbol kind changed from a non-type symbol to a type symbol",
    "[symbol-kind-changed] Export "F" symbol has changed (namespace, class, interface, or function).",
  ],
  "minChangeType": "major",
}
`;

exports[`Test case "symbol-add-export-value" 1`] = `
{
  "messages": [
    "[symbol-kind-changed] Export "F" symbol kind changed from a non-value symbol to a value symbol",
    "[symbol-kind-changed] Export "F" symbol has changed (namespace, class, interface, or function).",
  ],
  "minChangeType": "major",
}
`;

exports[`Test case "symbol-class-to-interface" 1`] = `
{
  "messages": [
    "[symbol-kind-changed] Export "Foo" symbol kind changed from a value symbol to a non-value symbol",
    "[symbol-kind-changed] Export "Foo" symbol has changed (namespace, class, interface, or function).",
  ],
  "minChangeType": "major",
}
`;

exports[`Test case "symbol-class-to-type" 1`] = `
{
  "messages": [
    "[symbol-kind-changed] Export "Foo" symbol kind changed from a value symbol to a non-value symbol",
  ],
  "minChangeType": "major",
}
`;

exports[`Test case "symbol-class-to-value-only" 1`] = `
{
  "messages": [
    "[symbol-kind-changed] Export "Foo" symbol kind changed from a type symbol to a non-type symbol",
    "[user-constructible-required-properties-added] New required property "prototype" has been added to "Foo"",
  ],
  "minChangeType": "major",
}
`;
